src/test/groovy/graphql/GraphQLTest.groovy

```diff
@@ -45,17 +47,19 @@ class GraphQLTest extends Specification {
                 .field(
                 newFieldDefinition()
                         .name("id")
+                        .dataFetcher(new MapDataFetcher("id"))
```

Comment:
Question: Is this interchangeable with a FieldDataFetcher? I'm basically struggling to understand the use case of a MapDataFetcher/PropertyDataFetcher over a FieldDataFetcher. I get that an object has fields that can be "fetched". Those fields could potentially be fetched as properties too using their getters and since its all a json object one could possibly serialize it to a map (not sure how) and cast the object to a map. But when would a user choose one over the other. Sorry if I'm missing the obvious...